tests/benchmarks/qasm/benchmark_downloader.py

```
@@ -0,0 +1,77 @@
+
+        raise ValueError(f"Unknown benchmark file: {filename}")
+
+    def _fetch_remote_file(self, filename: str, file_info: Dict) -> Path:
+        """Fetch a remote benchmark file into the cache and return its path."""
+        cached_path = self.cache_dir / filename
+        if cached_path.exists():
+            return cached_path
+
+        url = file_info["url"]
+        try:
+            with urllib.request.urlopen(url) as response:
```

Ruff (0.16.1)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)